invenio_alma/services/alma.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 Graz University of Technology.
+#
+# invenio-alma is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+
+"""Alma Service."""
+
+import requests
+from lxml import etree
+
+from .base import BaseService
+from .errors import AlmaRESTException
+
+
+class AlmaRESTService(BaseService):
+    """Alma REST service class."""
+
+    @staticmethod
+    def get(url):
+        """Alma rest api get request.
+
+        :param url (str): url to api
+
+        :raises AlmaRESTException if request was not successful
+
+        :return str: response content
+        """
+        response = requests.get(url, headers={"accept": "application/xml"})
+        if response.status_code >= 400:
+            raise AlmaRESTException(code=response.status_code, msg=response.text)
+        return response.text
+
+    @staticmethod
+    def put(url, data):
+        """Alma rest api put request.
+
+        :param url (str): url to api
+        :param data (str): payload
+
+        :raises AlmaRESTException if request was not successful
+
+        :return str: response content
+        """
+        response = requests.put(
+            url,
+            data,
+            headers={"content-type": "application/xml", "accept": "application/xml"},
+        )
+        if response.status_code >= 400:
+            raise AlmaRESTException(code=response.status_code, msg=response.text)
+        return response.text
+
+
+class AlmaService(AlmaRESTService):
+    """Alma service class."""
+
+    @staticmethod
+    def _extract_almarecord(data):
+        """Extract record from request.
+
+        :param data (str): result list
+
+        :return str: extracted record
```

Comment:
return type is not a string, is it a lxml Element?

Reply:
Done